### PR TITLE
8309602: update JVMTI history table for jdk 21

### PR DESCRIPTION
**Pull request by sspitsyn**
This is a minor update of the `jvmti.xml` file.
The JVM TI history table needs to be updated to list:
 - Virtual threads finalized to be a permanent feature.
 - Agent start-up in the live phase now specified to print a warning.

The JVM TI history table has no normative changes. This update does not need a CSR.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309602](https://bugs.openjdk.org/browse/JDK-8309602): update JVMTI history table for jdk 21 (**Bug** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [a9e4f8eb](https://git.openjdk.org/jdk/pull/14352/files/a9e4f8eb7c713b02c9c7782ffab8050a9251f59b)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [11db4f4f](https://git.openjdk.org/jdk/pull/14352/files/11db4f4f4f66d86d6bbf6af46e372c6e22c17eda)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14352/head:pull/14352` \
`$ git checkout pull/14352`

Update a local copy of the PR: \
`$ git checkout pull/14352` \
`$ git pull https://git.openjdk.org/jdk.git pull/14352/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14352`

View PR using the GUI difftool: \
`$ git pr show -t 14352`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14352.diff">https://git.openjdk.org/jdk/pull/14352.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14352#issuecomment-1580733315)

### CODE REVIEW

File: src/hotspot/share/prims/jvmti.xml [hunk 1]
```diff
@@ -15456,7 +15456,7 @@ typedef void (JNICALL *jvmtiEventVMInit)
       Minor update to deprecate Heap functions 1.0.
   </change>
   <change date="27 April 2022" version="19.0.0">
-      Support for virtual threads:
+      Preview feature - Support for virtual threads:
```
Comment (AlanBateman) [inline src/hotspot/share/prims/jvmti.xml:15459]: Maybe "Support for virtual threads as preview feature" or "Support for Virtual Threads (Preview)".

**Review by AlanBateman — APPROVED** [review]
A small suggestion for the 19.0.0 note, otherwise good.

File: src/hotspot/share/prims/jvmti.xml [hunk 2]
```diff
@@ -15468,6 +15468,10 @@ typedef void (JNICALL *jvmtiEventVMInit)
        - Add new error code:
          - JVMTI_ERROR_UNSUPPORTED_OPERATION
   </change>
+  <change date="7 June 2023" version="21.0.0">
+      Virtual threads finalized to be a permanent feature.
+      Agent start-up in the live phase now specified to print a warning.
+  </change>
```
Comment (AlanBateman) [inline src/hotspot/share/prims/jvmti.xml:15474]: This is just the history text, no normative changes, shouldn't need a CSR.

File: src/hotspot/share/prims/jvmti.xml (hunk 1 above)
Comment (sspitsyn) [reply src/hotspot/share/prims/jvmti.xml]: Thanks. Fixed now.

File: src/hotspot/share/prims/jvmti.xml (hunk 2 above)
Comment (sspitsyn) [reply src/hotspot/share/prims/jvmti.xml:15474]: Okay, thanks. Updated the description.

**Review by irisclark — APPROVED** [review]
Thanks!

**Comment by sspitsyn** [comment]
Alan and Iris, thank you for review!

**Pull request closed without merging**